lib/raxx/server.ex

```diff
@@ -183,14 +156,65 @@ defmodule Raxx.Server do
   """
   @callback handle_info(any(), state()) :: next
 
-  defmacro __using__(options) do
+  use Raxx.View, template: "server.html.eex", arguments: [:module]
+
+  defmacro __using__(_options) do
     quote do
       @behaviour unquote(__MODULE__)
+      import Raxx
 
-      use Raxx.NotFound, unquote(options)
+      @impl unquote(__MODULE__)
+      def handle_head(_request, _state) do
+        response(:not_found)
+        |> Raxx.Server.render(__MODULE__)
+      end
 
-      import Raxx
-      alias Raxx.{Request, Response}
+      @impl unquote(__MODULE__)
+      def handle_data(data, state) do
+        import Logger
+        Logger.warn("Received unexpected data: #{inspect(data)}")
+        {[], state}
+      end
+
+      @impl unquote(__MODULE__)
+      def handle_tail(trailers, state) do
+        import Logger
+        Logger.warn("Received unexpected trailers: #{inspect(trailers)}")
+        {[], state}
+      end
+
+      @impl unquote(__MODULE__)
+      def handle_info(message, state) do
+        import Logger
+        Logger.warn("Received unexpected message: #{inspect(message)}")
+        {[], state}
+      end
+
+      defoverridable unquote(__MODULE__)
+      @before_compile unquote(__MODULE__)
+    end
+  end
+
+  # DEBT Remove this for 1.0 release
+  defmacro __before_compile__(_env) do
+    quote do
+      try do
+        defoverridable handle_request: 2
+
+        if !Module.get_attribute(__MODULE__, :raxx_safe_server) do
+          %{file: file, line: line} = __ENV__
+
+          :elixir_errors.warn(__ENV__.line, __ENV__.file, """
+          The server `#{inspect(__MODULE__)}` implements `handle_request/2.
+              In place of `use Raxx.Server` try `use Raxx.SimpleServer.`
+              The behaviour Raxx.Server changes in release 0.17.0, see CHANGELOG for details.
+          """)
+        end
+      rescue
+        ArgumentError ->
+          # Do nothing handle_request has not been defined
```

Comment:
I love how well this works 👏 

I think `if Module.defines?(__MODULE__, {:handle_request, 2})` would work instead of the try/rescue, or would it have some problems?

Reply:
Did not know about this function.
Yes it works. I had tried `__MODULE__.module_info` but that is only available once compilation has finished